File: week1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);

int main(void)
{
    int cents = get_cents();

    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;
    int pennies = cents;

    int coins = quarters + dimes + nickels + pennies;

    printf("%i\n", coins);
}

int get_cents(void)
{
    int cents;

    do
    {
        cents = get_int("Change owed: ");
    } while (cents < 0);

    return cents;
}

int calculate_quarters(int cents)
{
    int quarters;
    quarters = cents / 25;

    return quarters;
}

int calculate_dimes(int cents)
{
    int dimes;
    dimes = cents / 10;

    return dimes;
}

int calculate_nickels(int cents)
{
    int nickels;
    nickels = cents / 5;

    return nickels;
}

File: week6/sentimental-cash/cash.py
def main():

    cents = get_cents()

    quarters = calculate_quarters(cents)
    cents -= quarters * 25

    dimes = calculate_dimes(cents)
    cents -= dimes * 10

    nickels = calculate_nickels(cents)
    cents -= nickels * 5

    pennies = cents

    coins = int(quarters + dimes + nickels + pennies)
    print(coins)

def get_cents():

    dollars = -1

    while dollars < 0:

        try:
            dollars = float(input("Change owed: "))
        except ValueError:
            print("Your input has to be a positive number")

    cents = dollars * 100

    return cents

def calculate_quarters(cents: int) -> int:
    quarters = cents // 25

    return quarters

def calculate_dimes(cents: int) -> int:
    dimes = cents // 10

    return dimes

def calculate_nickels(cents: int) -> int:
    nickels = cents // 5

    return nickels

if __name__ == "__main__":
    main()
